cdap-app-fabric/src/main/java/io/cdap/cdap/gateway/handlers/JsonListResponder.java

```diff
@@ -0,0 +1,106 @@
+/*
+ * Copyright © 2022 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package io.cdap.cdap.gateway.handlers;
+
+import com.google.common.base.Throwables;
+import com.google.gson.Gson;
+import com.google.gson.stream.JsonWriter;
+import io.cdap.http.ChunkResponder;
+import io.cdap.http.HttpResponder;
+import io.netty.handler.codec.http.HttpResponseStatus;
+
+import java.io.IOException;
+import java.io.Writer;
+import java.nio.ByteBuffer;
+import java.nio.CharBuffer;
+import java.nio.charset.StandardCharsets;
+
+/**
+ * JsonListResponder - parent class for list responders for streaming APIs
+ */
+public abstract class JsonListResponder {
+
+  protected final HttpResponder responder;
+  protected final JsonWriter jsonWriter;
+  protected Gson gson;
+  protected ChunkResponder chunkResponder;
+  protected boolean firstObjectReceived;
+
+  /**
+   * Constructs an instance of JsonListResponder
+   * @param gson - Instance of gson object used to do Json conversion
+   * @param responder - the HttpResponder that will be used in the streaming application
```

Comment:
Be more specific how it's going to be used. Say that json response will be written to it